src/utilities/manifest-utilities.ts

```diff
@@ -1,5 +1,100 @@
+'use strict';
+
 import * as core from '@actions/core';
-import { isEqual } from "./utils";
+import * as kubectlutility from '../kubectl-util';
+import * as io from '@actions/io';
+import { isEqual } from "./utility";
+
+export function getManifestFiles(manifestFilePaths: string[]): string[] {
+    if (!manifestFilePaths) {
+        core.debug('file input is not present');
+        return null;
+    }
+
+    return manifestFilePaths;
+}
+
+export async function getKubectl(): Promise<string> {
+    try {
+        return Promise.resolve(io.which('kubectl', true));
+    } catch (ex) {
+        return kubectlutility.downloadKubectl(await kubectlutility.getStableKubectlVersion());
+    }
+}
+
+export function createKubectlArgs(kinds: Set<string>, names: Set<string>): string {
+    let args = '';
+    if (!!kinds && kinds.size > 0) {
+        args = args + createInlineArray(Array.from(kinds.values()));
+    }
+
+    if (!!names && names.size > 0) {
+        args = args + ' ' + Array.from(names.values()).join(' ');
+    }
+
+    return args;
+}
+
+export function getDeleteCmdArgs(argsPrefix: string, inputArgs: string): string {
+    let args = '';
+
+    if (!!argsPrefix && argsPrefix.length > 0) {
+        args = argsPrefix;
+    }
+
+    if (!!inputArgs && inputArgs.length > 0) {
+        if (args.length > 0) {
+            args = args + ' ';
+        }
+
+        args = args + inputArgs;
+    }
+
+    return args;
+}
+
+/*
+    For example,
+        currentString: `image: "example/example-image"`
+        imageName: `example/example-image`
+        imageNameWithNewTag: `example/example-image:identifiertag`
+
+    This substituteImageNameInSpecFile function would return
+        return Value: `image: "example/example-image:identifiertag"`
+*/
+
+export function substituteImageNameInSpecFile(currentString: string, imageName: string, imageNameWithNewTag: string) {
+    if (currentString.indexOf(imageName) < 0) {
+        core.debug(`No occurence of replacement token: ${imageName} found`);
+        return currentString;
+    }
+
+    return currentString.split('\n').reduce((acc, line) => {
+        const imageKeyword = line.match(/^ *image:/);

```

Comment:
parse yaml and replace instead of string replace. I have opened work item for this.